### PR TITLE
Language switching controls and locale consistency

### PR DESCRIPTION
**Pull request by jiteshjha**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] tests pass -- `rake test`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request are descriptively named
* [x] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer
* [x] `development.sqlite.example` has been updated if any database migrations were added

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

---

@jywarren @ebarry What do you think about the location of language switching dropdown? Let me know your thoughts.

### CODE REVIEW

**Comment by jywarren** [comment]
Hi jitesh - I won't have a chance to merge this until Tuesday morning, but
do you think you're ready to write a few sentences of documentation on the
switch and how to use it? And/or how to do translations?

Thanks!

**Comment by jiteshjha** [comment]
Thanks for your input! By documentation on the switch, do you mean instructions for the users or just comments in the code?

**Comment by jywarren** [comment]
Can you add a short section to the README.md file about internationalization? We can start with a brief thing, link to any relevant documentation on other sites (rather than copy too much text in) and that's a good start.

**Comment by jywarren** [comment]
Hi, Jitesh - I think I'd like to wait a little on this. Do you think you could upload a couple screenshots of the the dropdown so everyone can review without having to boot it up? And do you think the README addition is possible? 

Thanks! I may be able to merge this on Wed or Thurs if you're ready by then.

**Comment by jywarren** [comment]
Mainly it's a design question at this point, the code looks great though.

**Comment by jiteshjha** [comment]
screenshot from 2016-06-14 21 50 43

**Comment by jiteshjha** [comment]
screenshot from 2016-06-14 21 50 52

**Comment by jiteshjha** [comment]
@jywarren I made an addition to the README, and uploaded the pictures depicting Language dropdown in action. What do you suggest?

**Comment by jywarren** [comment]
Thanks, the README addition is good, but can you add a new section like:

````markdown
### Internationalization
````

and mention where the actual translation files can be found and to write to plots-dev@googlegroups.com to ask about adding new languages?

As for the design part, do you think you could put language as a sub-menu of the "Login" window, as part of the window that shows your username (for logged-in users) and also as a dropdown in the footer?

**Comment by jywarren** [comment]
Oh great -- can you post screenshots of the new locations?

**Comment by jiteshjha** [comment]
screenshot from 2016-06-15 01 00 46
screenshot from 2016-06-15 01 00 59
screenshot from 2016-06-15 01 01 14

**Comment by jywarren** [comment]
This looks good -- I'm going to merge it. But can you look for an "internationalization" icon in fontawesome that you can place before the word "language" -- because otherwise non-english-speakers may not know to click there.

File: app/views/layouts/_footer.html.erb [hunk 1]
```diff
@@ -25,6 +25,15 @@
     <p><%= t('layout._footer.different_licenses') %></p>
     <p><%= raw t('layout._footer.font_use', :url => 'http://www.theleagueofmoveabletype.com/junction') %></p>
     <p><%= raw t('layout._footer.server_donation', :url => 'http://rackspace.com/') %></p>
+    <ul class="dropdown">
+      <a class="dropdown-toggle" data-toggle="dropdown" href="#">
+        <%= t('layout._footer.language.language_title') %> <b class="caret"></b>
+      </a>
+      <ul class="dropdown-menu">
+        <li><%= link_to t('layout._footer.language.english'), change_locale_path(:en) %></li>
```
Comment (jywarren) [inline app/views/layouts/_footer.html.erb]: Can you reduce duplicated code here (and in the two matching sections) by iterating over a list of available languages stored or set at some central location?

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -9,6 +9,17 @@
 <% end %>
 <p class="facebook-summary" style="display:none;"><%= t('layout._header.summary') %></p>
 
+<script>
```
Comment (jywarren) [inline app/views/layouts/_header.html.erb]: If this is on every page, can it go in `app/assets/javascripts/setup.js` instead of inline?

**Comment by jywarren** [comment]
Hi, sorry, i'm going to make a few more requests before merging:

* can you rebase it over the most recent master?
* I left a couple inline requests
* can you explain `set_locale` vs. `change_locale`, and if it makes sense, leave a brief comment above each explaining the difference?
* can you think about what other failure points we should test?

I hope these are small enough that we can merge promptly afterwards; thanks for your hard work and your patience.

**Pull request closed without merging**